wallet/db_sqlite3.c

```diff
@@ -489,22 +491,26 @@ static char **prepare_table_manip(const tal_t *ctx,
 	parts = tal_strsplit(ctx, bracket + 1, ",", STR_EMPTY_OK);
 
 	/* Turn off foreign keys first. */
-	sqlite3_prepare_v2(db->conn, "PRAGMA foreign_keys = OFF;", -1, &stmt, NULL);
+	sqlite3_prepare_v2(wrapper->conn, "PRAGMA foreign_keys = OFF;", -1, &stmt, NULL);
 	if (sqlite3_step(stmt) != SQLITE_DONE)
 		goto sqlite_stmt_err;
 	sqlite3_finalize(stmt);
 
 	cmd = tal_fmt(tmpctx, "ALTER TABLE %s RENAME TO temp_%s;",
 		      tablename, tablename);
-	sqlite3_prepare_v2(db->conn, cmd, -1, &stmt, NULL);
+	sqlite3_prepare_v2(wrapper->conn, cmd, -1, &stmt, NULL);
 	if (sqlite3_step(stmt) != SQLITE_DONE)
 		goto sqlite_stmt_err;
 	sqlite3_finalize(stmt);
 
+	/* Make sure we do the same to backup! */
+	replicate_statement(wrapper, "PRAGMA foreign_keys = OFF;");
+	replicate_statement(wrapper, cmd);
```

Comment:
***WAIT A MINUTE!***  If you are calling `replicate_statement` directly here, that means you are *not* calling `db_sqlite3_changes_add`.  And that means the queries are not propagated to `db_write` hook, meaning it is not propagated to *all* backup methods.  Would this not cause problems if the user were using the `backup` plugin?  Some queries sent to the local copy are not sent to the `backup` plugin?

Reply:
Fortunately it seems we do not actually use `delete_columns` or `rename_columns` *yet*, so we can still fix this, probably by rewriting in terms of `db_sqlite3_exec` (which gets the proper unwrapping of `db->conn` *and* replication to *all* backup processes we have, for free).